### DISCUSSION TITLE
Battery state not shown when parked

### DISCUSSION THREAD

**Discussion by dawiinci** [Thomas55555/husqvarna_automower#243] [Ideas]
The state of charging is not shown in the latest version. Instead it shows parked and the battery entity shows not charging. This seems to be an issue of the API itself, because in the App it is also like that. 

Is there any way to handle this in the plugin? My workaround for now is to check if parked and below 100% battery.

**Comment by Thomas55555** [comment]
For me it looks right. When it is docked and the battery is not 100% then charging is on, If it is still docked and the battery is full, the charging is off. Isn't it the same behaviour like you have.
![DeepinBildschirmfoto_Bereich auswählen_20220528172446](https://user-images.githubusercontent.com/59625598/170831936-0c51ba74-4bae-4cae-81ad-96b3433783f9.png)
I actually only want to represent the information of the API.
Is your charging binary sensor sometimes on, or never?

**Comment by Thomas55555** [comment]
Ah okay, now I see the problem. If the mower is not in a schedule, and "Parked until further notice". Then the `charging` is overwritten.
![DeepinBildschirmfoto_Bereich auswählen_20220528180103](https://user-images.githubusercontent.com/59625598/170833269-267d7f65-bf75-4acf-8318-0576a6a01934.png)
I don't think, it's the right way, to do handicrafts in this integration. Maybe it's better, if Husqvarna changes it. Maybe you can ask them?

**Reply by dawiinci** [reply]
Makes sense, I will check with them.

**Reply by dawiinci** [reply]
No response yet. But I have a workaround:

If docked and level is below 100 and status was cleaning in the last hour then it is charging. Works perfectly. 

I do this in HA.
